staging/src/k8s.io/apiserver/pkg/util/flowcontrol/debug/dump.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	"time"
+)
+
+// QueueSetDump is an instant dump of queue-set.
+type QueueSetDump struct {
+	Queues    []QueueDump
+	Waiting   int
+	Executing int
+}
+
+// QueueDump is an instant dump of one queue in a queue-set.
+type QueueDump struct {
+	Requests          []RequestDump
+	VirtualStart      float64
+	ExecutingRequests int
+}
+
+// RequestDump is an instant dump of one requests pending in the queue.
+type RequestDump struct {
+	MatchedFlowSchema string
+	FlowDistinguisher string
+	ArriveTime        time.Time
+	StartTime         time.Time
+	// request details
+	Verb              string
+	UserName          string
+	TargetNamespace   string
+	TargetName        string
+	TargetAPIGroup    string
+	TargetAPIVersion  string
+	TargetResource    string
+	TargetSubresource string
```

Comment:
What about non-resource requests?

Reply:
good point, i replace w/ a column named `APIPath` which already covers request attributes like`Namespace`, `Name`, `APIGroup`, etc..

Reply:
Mostly, but not in all cases.  For example, in a LIST request the target object name is extracted from the URL's path if that contains the name and otherwise from the query if that contains the name (which it can, as a field selector on `metadata.name`).

Reply:
Also, it might be convenient for consumers to have the parsed pieces even when they are all in the URL.
So I suggest restoring the old columns as well as keeping the path column.